.github/workflows/snapshot.yml

```diff
@@ -9,287 +9,330 @@ on:
-        if: github.event_name == 'pull_request'
-        id: tests
-        uses: gradle/gradle-build-action@v2
-        env:
-          GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER }}
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY }}
+      - name: "🐘 Setup Gradle"
+        uses: gradle/actions/setup-gradle@v4
         with:
-          arguments: check
-      - name: Run Build
-        if: github.event_name == 'push'
+          develocity-access-key: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
+      - name: "🔨 Run Build"
         id: build
-        uses: gradle/gradle-build-action@v2
         env:
-          GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER }}
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY }}
-        with:
-          arguments: build
-      - name: Verify CLI
-        if: steps.build.outcome == 'success' && github.event_name == 'push'
+          DEVELOCITY_BUILD_CACHE_NODE_USER: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER }}
+          DEVELOCITY_BUILD_CACHE_NODE_KEY: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY }}
+        run: ./gradlew build
+      - name: "✅ Verify CLI"
+        if: success()
         run: |
           cp grails-cli/build/distributions/grails-cli-*.zip cli.zip
           unzip cli -d tmp
           mv tmp/grails-cli-* tmp/cli
           ./tmp/cli/bin/grails --version
-      - name: Publish to Sonatype OSSRH
+      - name: "📤 Publish to Sonatype OSSRH"
         id: publish
-        if: steps.build.outcome == 'success' && github.event_name == 'push'
-        uses: gradle/gradle-build-action@v2
+        if: success() && github.event_name == 'push' && matrix.java == '11'
         env:
+          DEVELOCITY_BUILD_CACHE_NODE_USER: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER }}
+          DEVELOCITY_BUILD_CACHE_NODE_KEY: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY }}
           SONATYPE_USERNAME: ${{ secrets.SONATYPE_USERNAME }}
           SONATYPE_PASSWORD: ${{ secrets.SONATYPE_PASSWORD }}
-          GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GRADLE_ENTERPRISE_ACCESS_KEY }}
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_USER }}
-          GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY: ${{ secrets.GRADLE_ENTERPRISE_BUILD_CACHE_NODE_KEY }}
-        with:
-          arguments: publishToSonatype
+        run: ./gradlew publishToSonatype
   deploy:
-    name: Deploy To Google Cloud Run
+    # Snapshots should only be deployed from the latest release branch
+    # as it updates the snapshot version on https://start.grails.org.
+    if: false # github.event_name != 'pull_request'
```

Comment:
Is there a reason we're leaving the deploy step here if we're going to hard code it to false?  From what I can tell you moved it to the release.yml step, how do we ensure the latest 7.0.0 build gets built if it's not released yet?

Reply:
> Is there a reason we're leaving the deploy step here if we're going to hard code it to false?

Well, I could remove it instead but I thought it was self-documenting having it there. And if you, for some reason, did need to deploy, you and just flip the switch.

Would you rather have me remove it?

> From what I can tell you moved it to the release.yml step

 I don't think I moved it to `release.yml`. It was already there.

> how do we ensure the latest 7.0.0 build gets built if it's not released yet?

I'm not sure I understand the question. This PR is for the 6.0.x branch. I am planning to merge it up to 6.1.x and 6.2.x making changes as necessary. The 7.0.x branch will not get any changes to GCR publishing at the moment.

Reply:
I understand now, lets leave it?